Add support for Python context managers.

teaearlgraycold opened the pull request:
I added two methods to the `Halo` class: `__enter__()` and `__exit__()`. These allows spinners to be used through Python's `with` statement like so:

```python
from halo import Halo

with Halo({'text': 'Loading', 'spinner': 'dots'}):
    # Run time consuming work here
```

---

As an aside, the way that keyword parameters are passed to the Halo class is non-Pythonic. You should just use normal kwargs with default values.

teaearlgraycold commented:
It looks like my editor (atom) removed some stray white-space.

ManrajGrover commented:
Hi @teaearlgraycold, this was on my list to implement. Thank you for sending a PR. I'll review the code soon. Those docstrings were auto generated, hence the spacing. No worries.

I do agree the keyword params are Non-Pythonic. I should use `*args` and `**kwargs`. I plan to fix that soon. If you want, you can send a PR for that too. 😄

teaearlgraycold commented:
>If you want, you can send a PR for that too.

Currently working on that.

I don't see the need for using the splat operator. All of the kwargs can be explicitly stated.

ManrajGrover commented on halo/halo.py:
```diff
@@ -65,12 +65,24 @@ def __init__(self, options={}):
+        """
+        return self.start()
+
+    def __exit__(self, type, value, traceback):
```
We should document `__exit__` too. Also `return self.stop()`.

ManrajGrover commented:
@teaearlgraycold With recent PR merged which moves from dictionary based param for constructor to keyword arguments, this PR now has merge conflicts. Please resolve them. :)

teaearlgraycold commented:
Updated. I added a docstring to `__exit__()` but it seems that `__exit__()` generally doesn't return anything from what I can see in context manager code online.

ManrajGrover commented:
@teaearlgraycold Looks correct to me. Also, since the API now uses keyword args, we need to fix tests too.
